### PR TITLE
Fix bookmarks toolbar overflow indicator display

### PR DESCRIPTION
**Pull request by GreenRecycleBin**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by GreenRecycleBin** [comment]
@bsclifton Could you take a look?

**Comment by cezaraugusto** [comment]
@GreenRecycleBin could you pls rebase?

**Comment by GreenRecycleBin** [comment]
@cezaraugusto [A recent commit](https://github.com/brave/browser-laptop/commit/705b4e49f12f034b3115c534eaae92a60a479c7f#diff-acfcdd117a10c840d0043b12dacbab9bR72) fixed the regression introduced by [an earlier commit] (https://github.com/brave/browser-laptop/commit/85380251cf8056910b1f8552f7553f68a30d0f40#diff-49b40c07650f6f7cedce3f608d031b54R112). I've removed one of my previous commit that did the same thing and rebased the other onto `master`.

**Comment by GreenRecycleBin** [comment]
@cezaraugusto [A recent commit](https://github.com/brave/browser-laptop/commit/705b4e49f12f034b3115c534eaae92a60a479c7f#diff-acfcdd117a10c840d0043b12dacbab9bR72) fixed the regression introduced by [an earlier commit](https://github.com/brave/browser-laptop/commit/85380251cf8056910b1f8552f7553f68a30d0f40#diff-49b40c07650f6f7cedce3f608d031b54R112). I've removed one of my previous commit that did the same thing and rebased the other onto master.

**Comment by cezaraugusto** [comment]
you're in the right track btw but calculation still needs some tweaks. let me know if you need a hand

**Comment by bsclifton** [comment]
Hey there @GreenRecycleBin- it's been a while, I just wanted to follow up with you on this one

@cezaraugusto had left some feedback (above).  Have you gotten a chance to look at it again?  Please let us know if you need a hand 😄

**Comment by GreenRecycleBin** [comment]
I noticed [`bookmark-item-font-size` in `styleValues.js`](https://github.com/brave/browser-laptop/blob/be93d995949cd6dd9cc7dc5cdb73dd9c7dfa1e64/app/common/constants/styleValues.js#L10) is different from [the corresponding one in `bookmarksToolbar.less`](https://github.com/brave/browser-laptop/blob/be93d995949cd6dd9cc7dc5cdb73dd9c7dfa1e64/less/bookmarksToolbar.less#L12). Furthermore, `bookmarkToolbarButton.js` is using [a different `font-size`](https://github.com/brave/browser-laptop/blob/be93d995949cd6dd9cc7dc5cdb73dd9c7dfa1e64/app/renderer/components/bookmarks/bookmarkToolbarButton.js#L321) from the above two.

The first two obviously need to be in sync according to [this comment](https://github.com/brave/browser-laptop/blob/be93d995949cd6dd9cc7dc5cdb73dd9c7dfa1e64/app/common/constants/styleValues.js#L5) and [the other one](https://github.com/brave/browser-laptop/blob/be93d995949cd6dd9cc7dc5cdb73dd9c7dfa1e64/less/bookmarksToolbar.less#L6).

Should we keep these bookmark related styles in sync across these three files? If so, what's the best way to do that? Thank you.

**Comment by cezaraugusto** [comment]
@GreenRecycleBin we're in the process of moving things out of LESS and make use of Aphrodite. Styles shared across components should live in `global.js` file. 

All LESS files and `styleValues.js` will both eventually be removed. I would suggest you looking which style is really applied for what you want and move them to `global.js`. Even if there's a dupe of global.js and LESS for bookmarks, that's ok. Global styles should be preferred and computed styles methods avoided.

You can find a good example in [bookmarkToolbarButton](https://github.com/brave/browser-laptop/blob/be93d995949cd6dd9cc7dc5cdb73dd9c7dfa1e64/app/renderer/components/bookmarks/bookmarkToolbarButton.js#L319). You can move `fontSize: '11px'` to global and do something like `fontSize: globalStyles.bookmarks.fontSize`. Then use that style in the calc you're doing. Repeat this pattern wherever needed. For code still in LESS, just copy the style property to `global.js` and consult that file to get the number. 

let me know if that makes sense for you

**Comment by GreenRecycleBin** [comment]
@cezaraugusto Could you take another look?

**Comment by GreenRecycleBin** [comment]
@cezaraugusto There were some conflicts with `origin/master`. I've resolved them. Could you take a look?

File: app/common/lib/bookmarkUtil.js [hunk 1]
```diff
@@ -71,57 +83,94 @@ let lastWidth
...
-  const fontSize = domUtil.getStyleConstants('bookmark-item-font-size')
-  const fontFamily = domUtil.getStyleConstants('default-font-family')
-  const chevronWidth = chevronMargin + fontSize
-  const margin = favicon && onlyFavicon ? 0 : bookmarkItemMargin
+  const margin = onlyFavicon ? 0 : bookmarkItemMargin
+
+  // Toolbar padding is only on the left
+  const toolbarPadding = parseInt(globalStyles.spacing.bookmarksToolbarPadding)
+
+  const overflowButtonWidth = 25
```
Comment (cezaraugusto) [inline app/common/lib/bookmarkUtil.js]: I know it was like this before but could you please add this to `globalStyles` as well?

File: app/renderer/components/bookmarks/bookmarkToolbarButton.js [hunk 2]
```diff
@@ -304,10 +304,6 @@ class BookmarkToolbarButton extends React.Component {
 
 module.exports = ReduxComponent.connect(BookmarkToolbarButton)
 
-const bookmarkItemMaxWidth = '100px'
-const bookmarkItemPadding = '4px'
-const bookmarkItemMargin = '3px'
-const bookmarkItemChevronMargin = '4px'
 const bookmarkToolbarButtonDraggingMargin = '25px'
```
Comment (cezaraugusto) [inline app/renderer/components/bookmarks/bookmarkToolbarButton.js]: this could benefit from the above comment as well

**Comment by GreenRecycleBin** [comment]
@cezaraugusto Could you take another look?

If everything works, could you do `git rebase --interactive --autosquash` and merge? Thanks.

**Comment by bsclifton** [comment]
@GreenRecycleBin thanks for hanging in there and delivering the fix! 😄

**Pull request merged**